### PR TITLE
Fix the invalid spirv generation for matrix cast

### PR DESCRIPTION
**Pull request by kaizhangNV**
Spirv doesn't have instruction to do the float cast for the matrix type. So we have to convert the matrix row by row, and then construct them to a new matrix.

Update the unit test to make sure the cast won't miss any elements.

Also update the workflow to validate the spirv generation.

### CODE REVIEW

**Comment by jkwak-work** [comment]
Were you able to reproduce the validation error locally without upgrading the vulkan sdk version?

**Comment by jkwak-work** [comment]
This PR will close #4589.

**Comment by kaizhangNV** [comment]
> sdk

Yes, I can. And now the issue is gone with this change.

File: .github/workflows/windows-selfhosted.yml [hunk 1]
```diff
@@ -47,6 +47,10 @@ jobs:
       
     - name: test
       run: |
+        $ErrorActionPreference = "SilentlyContinue"
+        where.exe spirv-dis
+        spirv-dis --version
+        $env:SLANG_RUN_SPIRV_VALIDATION='1'
```
Comment (jkwak-work) [inline .github/workflows/windows-selfhosted.yml:53]: I don't think we need this change anymore.
The file "windows-selfhosted.yml" is delete at ToT.
The same functionality is implemented in ci.yml, which sets SLANG_RUN_SPIRV_VALIDATION correctly.

Comment (kaizhangNV) [reply .github/workflows/windows-selfhosted.yml:53]: Ok, I can remove this file from this change.

**Review by csyonghe — DISMISSED** [review]
Looks good to me.

**Comment by kaizhangNV** [comment]
I rebase on TOT and remove the `windows-selfhosted.yml`.

**Pull request merged**